pkg/manager/task_scan.go

```diff
@@ -81,7 +81,8 @@ func (t *ScanTask) scanScene() {
 	qb := models.NewSceneQueryBuilder()
 	scene, _ := qb.FindByPath(t.FilePath)
 	if scene != nil {
-		// We already have this item in the database, keep going
+		// We already have this item in the database, check for thumbnails,screenshots
+		t.checkScreenshots(scene.Checksum)
```

Comment:
There is already a `makeScreenshots` function that can be reused here. It checks if the screenshot files exist and only rebuild if they are not present.

I would do the following:

```
qb := models.NewSceneQueryBuilder()
scene, _ := qb.FindByPath(t.FilePath)

videoFile, err := ffmpeg.NewVideoFile(instance.FFProbePath, t.FilePath)
if err != nil {
	logger.Error(err.Error())
	return
}

t.makeScreenshots(*videoFile, scene.Checksum)

if scene != nil {
    // We already have this item in the database, keep going
    return
}
```

Then the new function isn't needed at all. Sorry for not picking this up earlier.

Reply:
https://github.com/stashapp/stash/commit/a67a45ce910f9dfd475e77ab58eb8fe3ac109ae6

it used to be like that but invoking ffprobe for every single file slowed down the scan process too much.

Reply:
In that case, I would change `makeScreenshots` to accept a pointer to a `VideoFile`. If this is nil, then create the `VideoFile`. Then line 85 will be `t.makeScreenshots(nil, checksum)`, and the existing call to `makeScreenshots` will be changed to pass the pointer. Basically want to avoid repeated code if possible.

Reply:
i wanted to avoid editing the function but i guess i'll just do that ... sounds good

Reply:
Should be ok now 

@WithoutPants have a look